### ISSUE TITLE
[Bug] Column-slice of a dynamic-pitch tensor passed orch→incore silently reads wrong data (stride densified)

### ISSUE THREAD

**Issue by YunjiQin** [hw-native-sys/pypto#2087]
### Component

Codegen (root cause spans op type-deduction + IR passes + PTO codegen — see Additional Context)

### Description

When an orchestration function column-slices a tensor whose **last dim (row pitch) is dynamic** (`pl.dynamic(...)`) and passes the slice into an incore kernel, the incore GM load **silently reads wrong data** — no error, no NaN, no diagnostic.

Root cause: the incore GM operand's stride is emitted as a **compile-time constant** baked into the generated `GlobalTensor<..., pto::Stride<...>>`. `kernel_entry` passes only `buffer.addr + start_offset` at runtime — never a stride. For a **static** pitch, the compiler stamps the true parent pitch (e.g. `Stride<...,128,1>`) and the result is correct. For a **dynamic** pitch there is no runtime channel to thread the pitch into the kernel, so codegen falls back to the **dense/contiguous** assumption `stride = inner load extent` (`Stride<...,64,1>`), i.e. it reads the `[32,64]` window as if it were contiguous. Only rows where the dense stride happens to equal the true pitch survive.

Note: the runtime `Tensor::view()` **does** compute the correct dynamic stride (`result.strides[i] = strides[i]`) and stores it on the Tensor struct — but the incore kernel never reads it.

The equivalent **static-shape** column slice is correct (verified: max diff 0.0). Only the dynamic-pitch case breaks.

### Steps to Reproduce

```python
import pypto.language as pl
import torch
from pypto.runtime import RunConfig

N = pl.dynamic("N")   # dynamic last dim (= row pitch) of the parent tensor

@pl.jit.incore
def copy_kernel(a: pl.Tensor, out: pl.Out[pl.Tensor]):
    t = pl.load(a, [0, 0], [32, 64])
    pl.store(t, [0, 0], out)
    return out

@pl.jit
def col_split_dyn(x: pl.Tensor[[32, N], pl.FP32],
                  out0: pl.Out[pl.Tensor], out1: pl.Out[pl.Tensor]):
    left  = pl.slice(x, [32, 64], [0, 0])
    right = pl.slice(x, [32, 64], [0, 64])   # column offset on dim 1
    out0 = copy_kernel(left, out0)
    out1 = copy_kernel(right, out1)
    return out0, out1

x = torch.randn(32, 128, dtype=torch.float32)
o0 = torch.zeros(32, 64); o1 = torch.zeros(32, 64)
# codegen-only inspection (no device/runtime needed):
col_split_dyn(x, o0, o1, config=RunConfig(save_kernels=True, codegen_only=True))
# Inspect kernels/aiv/copy_kernel.cpp -> the load GlobalTensor stride is
#   pto::Stride<2048, 2048, 2048, 64, 1>   (dense, WRONG)
# A static x: pl.Tensor[[32, 128], pl.FP32] instead bakes Stride<...,128,1> (correct).
```

Reproduces on `a2a3sim` (not hardware-specific). A full numeric run gives `left/right max diff ≈ 5.5 / 4.9` versus the expected `0.0`.

### Expected Behavior

The incore load reads the correct strided window (`left == x[:, 0:64]`, `right == x[:, 64:128]`), matching torch — `max diff ≈ 0`, identical to the static-pitch case. If dynamic strides cannot yet be supported, the compiler should **error loudly** rather than silently emit a dense stride.

### Actual Behavior

Silently wrong data, no error/warning:

```
left  max diff: 5.50   # expected 0.0
right max diff: 4.91
# generated kernels/aiv/copy_kernel.cpp:
pto::Stride<2048, 2048, 2048, 64, 1>   # dense row stride 64 — true parent pitch is 128
```

### Git Commit ID

f990c8d7c92484abde9ac0f674a044e7a9cc28c7

### NPU Kind

N/A (not hardware-specific)

### Host Platform

Linux (aarch64)

### Additional Context

Related: #2005 (silent GM data corruption), #631 (orchestration codegen for incore calls).

**Cross-layer trace (static works, dynamic breaks):**

- Slice type deduction drops the parent stride — `DeduceTensorSliceType` builds the result `TensorView` with empty stride (`src/ir/op/tensor_ops/memory.cpp:282-290`).
- `OptimizeOrchTensors` only stamps a `TensorView(stride=...)` for **statically provable** windows; for a symbolic pitch it bails (`ComputeRowMajorStrides` returns `{}` on any non-const dim, `src/ir/transforms/optimize_orch_tensors_pass.cpp:64-81`, In-param path `:1461-1557`; window gate `src/ir/transforms/utils/window_externalization.cpp:2674-2691`). The callee param stays a **bare** `pl.Tensor[[32,64]]`.
- `MaterializeTensorStrides` treats a bare/empty-stride TensorType as implicitly ND-packed and fills a **dense** stride from the type's own shape (`src/ir/transforms/materialize_tensor_strides_pass.cpp:87-107` via `BuildLogicalStridesFromLayout`, `include/pypto/ir/transforms/utils/tensor_view_semantics.h:72-125`).
- Codegen emits the dense stride: `EmitMakeTensorViews` canonical-ND fallback stamps `stride[rank-2] = shape[rank-1]` (`src/codegen/pto/pto_codegen.cpp:1016-1028`, gate `:973-974`; pre-emit `:806-826`). This is the single site that produces the dense `Stride<...,64,1>`.
- The `TensorViewCanonical` verifier does not catch it — symbolic stride relations pass under `relaxed_symbolic` (`tensor_view_semantics.h:206-281`).

**Fix direction (feasible; mirrors the existing dynamic-shape path):** thread the runtime stride the same way dynamic shapes / `valid_shape` scalars are threaded — the wrapper (`kernel_entry`) reads `tensor->strides[i]` and passes it as a separate scalar arg into the ptoas func, and codegen emits `GlobalTensor` with the ISA's dynamic stride slot. The ISA already supports this: `pto::DYNAMIC` (`= -1`), `Stride<...>` constructors that fill `DYNAMIC` slots from runtime `int64_t` values, and `GlobalTensor(data, shape, stride)` taking a runtime Stride (`runtime/build/pto-isa/include/pto/common/pto_tile.hpp`; existing use `Stride<1,1,1,DYNAMIC,1>` in `pto/npu/kernels/Pto_prefetch.hpp`). So the parent pitch must be promoted to a first-class dynamic dim that flows through the existing dynamic-dim → runtime-scalar materialization machinery, rather than being densified to a compile-time constant.
